### ISSUE TITLE
`react-redux` >= 8 does not work with RTK Query

### ISSUE THREAD

**Issue by gonzofish** [reduxjs/redux-toolkit#2715]
I recently was trying to using RTK Query with React and followed [the documentation laid out for using React with RTK|https://redux-toolkit.js.org/tutorials/quick-start#install-redux-toolkit-and-react-redux]. While this seems to work fine when just using RTK and `react-redux`, trying to use RTK Query causes a stack track that doesn't seem to recognize the `react-redux` `Provider`.

<details>
  <summary>React Redux version after following docs</summary>
  <pre>
    > npm ls react-redux
    my-app@0.0.0
    └─┬ myapp
    ├─┬ @reduxjs/toolkit@1.8.5
    │ └── react-redux@7.2.8
    └── react-redux@8.0.2
  </pre>
</details>

<details>
  <summary>Stack trace when using RTK Query <code>getQuery</code></summary>
  <pre>
The above error occurred in the <TempApp> component:

TempApp@http://localhost:3001/static/js/bundle.js:51:74
Provider@http://localhost:3001/static/js/bundle.js:3191:7

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://reactjs.org/link/error-boundaries to learn more about error boundaries. react-dom.development.js:20092
</pre>

<pre>
Uncaught Error: could not find react-redux context value; please ensure the component is wrapped in a <Provider>
    useReduxContext bundle.js:76149
    useStore bundle.js:76369
    Redux 3
    TempApp index.tsx:35
    React 2
    beginWork bundle.js:63093
    React 7
    performSyncWorkOnRoot bundle.js:66289
    React 5
    render bundle.js:70028
    tsx index.tsx:48
    factory react refresh:8
    Webpack 3
2 bundle.js:76149:11
</pre>
</details>

To fix this I just installed `react-redux@^7.0.0` and that brought it inline with `@reduxjs/toolkit`. At the least, could this be mentioned in the docs? I'm willing to a PR for the docs if it's ok.

**Comment by markerikson** [comment]
I don't know why React-Redux v7 would have been installed. We don't actually install it ourselves - we just list both v7 and v8 as peer dependencies:

https://github.com/reduxjs/redux-toolkit/blob/7c3129c46c173902801d4f6143cd9ef88b5ee553/packages/toolkit/package.json#L105-L108

What package manager did you install with? I'm assuming `npm`?

My guess is this is likely due to NPM's "let's automatically install peer deps" behavior possibly getting things wrong.

**Comment by gonzofish** [comment]
Yes, I did install using `npm`

**Comment by markerikson** [comment]
Yeah, first thing would be to make sure there's only one copy of React-Redux in the codebase.

**Comment by markerikson** [comment]
Closing because this seems to be a package manager issue.
